src/logical/chrono.rs

```diff
@@ -94,13 +121,58 @@ impl LogicalArrayType<NaiveTime> for NaiveTime {
 pub type NaiveTimeArray<const NULLABLE: bool = false, Buffer = crate::buffer::VecBuffer> =
     LogicalArray<NaiveTime, NULLABLE, Buffer, crate::offset::NA, crate::array::union::NA>;
 
+impl ArrayType<TimeDelta> for TimeDelta {
+    type Array<Buffer: BufferType, OffsetItem: OffsetElement, UnionLayout: UnionType> =
+    LogicalArray<Self, false, Buffer, OffsetItem, UnionLayout>;
+}
+
+impl ArrayType<TimeDelta> for Option<TimeDelta> {
+    type Array<Buffer: BufferType, OffsetItem: OffsetElement, UnionLayout: UnionType> =
+    LogicalArray<TimeDelta, true, Buffer, OffsetItem, UnionLayout>;
+}
+
+/// The number of nano seconds in a milli second.
+const NANOS_PER_MILLI: i64 = 1_000_000;
+/// The number of milli seconds in a second.
+const MILLI_SECONDS: i64 = 1_000;
+
+impl LogicalArrayType<TimeDelta> for TimeDelta {
+    type ArrayType = i64;
+
+    fn from_array_type(item: Self::ArrayType) -> Self {
+        let (secs, nano) = (item.div_euclid(MILLI_SECONDS),
+                            u32::try_from(item.rem_euclid(NANOS_PER_MILLI))
+                                .expect("i64 to u32 cast error"));
+        
+        Self::new(secs, nano).expect("out of range")
```

Comment:
We can use https://docs.rs/chrono/latest/chrono/struct.TimeDelta.html#method.nanoseconds to simplify:
```suggestion
        Self::nanoseconds(item)
```
This also means that we don't need the consts.

Reply:
I could have checked the fn definition :)